### PR TITLE
Remove non-working check from x509_get_subject_alt_name

### LINKED ISSUE #2802: Condition always false in x509_get_subject_alt_name
Comparing with zero could be slightly more efficient, but also it is easier to understand.

https://github.com/ARMmbed/mbedtls/blob/beec142010cd9f3a967a9a3a7b407b51841b542e/library/x509_crt.c#L648
It might be guessed that the intention was to check `end>=*p`, but the values are pointers, and thus unsigned.
In pointer arithmetics anything less than 1 is nothing but 0.
Hence the condition means `end == *p`,

In any case, the loop header ensures that `*p < end`.

### PR DESCRIPTION
**Pull request by irwir**
## Description
That fixes  #2802
The changed comment might also be changed by @hanno-arm in his PR.

## Status
**READY**

## Requires Backporting
NO  
Which branch?
Development

## Migrations
NO

### CODE REVIEW

File: library/x509_crt.c [hunk 1]
```diff
@@ -662,7 +658,7 @@ static int x509_get_subject_alt_name( unsigned char **p,
         }
 
         /*
-         * Check that the SAN are structured correct.
```
Comment (k-stachowiak) [inline library/x509_crt.c]: This change is outside of the PRs scope. I'd suggest not mixing in extra changes with the main ones. But this is far from being a blocker.

**Comment by irwir** [comment]
The state is `approved`, but `needs: review` tag is still present.
Does it need more than 2 reviews?

**Comment by gilles-peskine-arm** [comment]
This is a minor improvement that doesn't require backporting. It does require a changelog. Can you please rebase this PR on top of `development` and add a `.txt` file in `ChangeLog.d`? Suggested content:

```
Bugfix
   * Remove dead code in X.509 certificate parsing. Contributed by irwir in #2855.
```

**Comment by irwir** [comment]
> It does require a changelog.

Thanks.
Changelog entry was added as a separate commit.

**Comment by danh-arm** [comment]
Hi @irwir . Sorry this PR stalled. Unfortunately you're change log entry filename conflicts with another one. Can you use a name that uniquely identifies this PR? Thanks.

**Comment by irwir** [comment]
Altered the file name.
I expected that the entries were to be added automatically for every merged PR, then file names were irrelevant. But apparently this is (yet?) not so.

**Comment by danh-arm** [comment]
> Altered the file name.
Thanks.

> I expected that the entries were to be added automatically for every merged PR, then file names were irrelevant. But apparently this is (yet?) not so.

The entries are only collated at release time so they must all be unique between releases. I accept the guidance needs improvement.

**Pull request merged**